test/API-functional/device-spec.js

```diff
@@ -0,0 +1,126 @@
+/* eslint no-undef: "off" */
+const assert = require('assert').strict
+const StApi = require('../../lib/api')
+// User token for prod env
+const opts = {authToken: '0594f1d2-69e9-4809-a773-5fe7268e3f01'}
+const api = new StApi(opts)
+
+let params
+let devicesApi
+let deviceId
+accessToken = '8ede1b8e-890e-4995-9ee8-ea4d06d885f5'
+async function fetchContent() {
+	const [profiles, installApp, locations] = await Promise.all([api.deviceProfiles.list(), api.installedApps.list(), api.locations.list()])
+	params = {
+		locationId: locations.body.items[0].locationId,
+		label: 'Switch.model1',
+		profileId: profiles.body.items[0].id,
+		installedAppId: installApp.body.items[0].installedAppId,
+		externalId: 'externalID',
+		authToken: accessToken
+	}
+	devicesApi = new StApi(params)
+	// Creates a device
+	await devicesApi.devices.create(params).then(response => {
+		deviceId = response.body.deviceId
+		componentId = response.body.components[0].id
+		capabilityId = response.body.components[0].capabilities[0].id
+	}).catch(() => {})
+}
+
+describe('Devices API', function () {
+	this.timeout(10000)
+	before(async () => {
+		// Assuming a device profile has been created and an endpoint based c2c app installed in a location.
+		await fetchContent()
+	})
+
+	after('deleting the created device', () => {
+		devicesApi.devices.delete(deviceId)
+	})
+
+	it('list devices', done => {
```

Comment:
mocha seems to [discourage](https://mochajs.org/#arrow-functions) passing arrow functions because you lose the _mocha_ context. Just a future tip, as you don't seem to need that context here anyway.

Reply:
Yes, acknowledged. But when changing to a regular callback function , it throws following, hence went ahead with arrow func for now. 
![image](https://user-images.githubusercontent.com/38669105/55635391-2769f380-5786-11e9-8ea4-b1b9ecb0392c.png)